packages/angular-output-target/src/generate-angular-component.ts

```diff
@@ -1,132 +1,165 @@
+   * Angular does not complain about the inputs property. The output target
+   * uses the inputs property to define the inputs of the component instead of
+   * having to use the @Input decorator (and manually define the type and default value).
+   */
+  const output = `@ProxyCmp({${proxyCmpOptions.join(',')}\n})
 @Component({
-  ${directiveOpts.join(',\n  ')}
+  selector: '${tagName}',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: '<ng-content></ng-content>',
+  // eslint-disable-next-line @angular-eslint/no-inputs-metadata-property
+  inputs: [${formattedInputs}],
 })
-export class ${tagNameAsPascal} {`,
-    ];
-
-    lines.push('  protected el: HTMLElement;');
-    lines.push(`  constructor(c: ChangeDetectorRef, r: ElementRef, protected z: NgZone) {
+export class ${tagNameAsPascal} {
+  protected el: HTMLElement;
+  constructor(c: ChangeDetectorRef, r: ElementRef, protected z: NgZone) {
     c.detach();
-    this.el = r.nativeElement;`);
-    if (hasOutputs) {
-      lines.push(`    proxyOutputs(this, this.el, ['${outputs.map((output) => output.name).join(`', '`)}']);`);
+    this.el = r.nativeElement;${
+      hasOutputs
+        ? `
+    proxyOutputs(this, this.el, [${formattedOutputs}]);`
+        : ''
     }
-    lines.push(`  }`);
-    lines.push(`}`);
-
-    return lines.join('\n');
-  };
-
-function getProxyCmp(tagName: string, includeCustomElement: boolean, inputs: string[], methods: string[]): string {
-  const hasInputs = inputs.length > 0;
-  const hasMethods = methods.length > 0;
-
-  const proxMeta: string[] = [
-    `defineCustomElementFn: ${includeCustomElement ? 'define' + dashToPascalCase(tagName) : 'undefined'}`,
-  ];
-
-  if (hasInputs) proxMeta.push(`inputs: ['${inputs.join(`', '`)}']`);
-  if (hasMethods) proxMeta.push(`methods: ['${methods.join(`', '`)}']`);
-
-  return `@ProxyCmp({\n  ${proxMeta.join(',\n  ')}\n})`;
-}
+  }
+}`;
+
+  return output;
+};
+
+/**
+ * Sanitizes and formats the component event type.
+ * @param componentClassName The class name of the component (e.g. 'MyComponent')
+ * @param event The Stencil component event.
+ * @returns The sanitized event type as a string.
+ */
+const formatOutputType = (componentClassName: string, event: ComponentCompilerEvent) => {
+  /**
+   * The original attribute contains the original type defined by the devs.
+   * This regexp normalizes the reference, by removing linebreaks,
+   * replacing consecutive spaces with a single space, and adding a single space after commas.
+   */
+  return Object.entries(event.complexType.references)
+    .filter(([_, refObject]) => refObject.location === 'local' || refObject.location === 'import')
+    .reduce(
+      (type, [src, dst]) => {
+        const renamedType = `I${componentClassName}${type}`;
+        return renamedType
+          .replace(new RegExp(`^${src}$`, 'g'), `${dst}`)
+          .replace(new RegExp(`([^\\w])${src}([^\\w])`, 'g'), (v, p1, p2) => [p1, dst, p2].join(''));
```

Comment:
We might want to consider adding a one-liner comment here to explain what this does (or at least its intent):
```suggestion
          // capture all instances of the `src` field surrounded by non-word characters on each side, join them
          .replace(new RegExp(`([^\\w])${src}([^\\w])`, 'g'), (v, p1, p2) => [p1, dst, p2].join(''));
```